dts/bindings/misc/zephyr,flash-disk.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: Block storage device on flash partition
+
+compatible: "zephyr,flash-disk"
+
+include: base.yaml
+
+properties:
+    partition:
+      type: phandle
+      required: true
+      description: |
+        Backing storage flash map partition.
+
+    disk-name:
+      type: string
+      required: true
+      description: |
+        Disk name.
+
+    sector-size:
+      type: int
+      required: false
+      default: 512
+      description: |
+        Emulated block device sector size in bytes.
+
+    erase-size:
```

Comment:
Property name should be `buffer-size` or `cache-size` or `foobar-size`, but not `erase-size`, it is very confusing and one assumes that it is the direct property of a raw flash device. I would not touch Kconfig option DISK_ERASE_BLOCK_SIZE in this PR.

Reply:
Now it is `cache-size`,  albeit personally I would prefer `buffer-size`.